paddle/fluid/operators/listen_and_serv_op.cc

```diff
@@ -373,6 +381,29 @@ from send_op and send back variables to recv_op.
   }
 };
 
+bool SignalHandler::program_exit_flag_ = false;
+
+SignalHandler::BlockingQueueSet SignalHandler::blocking_queue_set_{};
+
+void SignalHandler::StopAndExit(int signal_num) {
+  VLOG(3) << "Catch interrupt signal: " << signal_num << ", program will exit";
+
+  program_exit_flag_ = true;
+
+  // awake all blocking queues
+  for (BlockingQueueSet::iterator iter = blocking_queue_set_.begin();
+       iter != blocking_queue_set_.end(); iter++) {
+    iter->get()->Push(
+        std::make_pair(std::string(LISTEN_TERMINATE_MESSAGE), nullptr));
+  }
+
+  exit(signal_num);
```

Comment:
`exit(0)` if success. Curious why need to call `exit`?

Reply:
According to this: https://softwareengineering.stackexchange.com/a/140487, exit() will call all deconstructors in all classes including std::thread and RpcServer, so just call exit is fine in most cases.